### PR TITLE
Added support for loading zip-packed components

### PR DESCRIPTION
**Pull request by Ark-kun**
The zip-packed components are supported in all `load_component` APIs:
`kfp.components.load_component`
`kfp.components.load_component_from_file`
`kfp.components.load_component_from_url`
`kfp.components.ComponentStore.load_component`

This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/931)

### CODE REVIEW

File: sdk/python/kfp/components/_components.py [hunk 1]
```diff
@@ -113,6 +113,31 @@ def load_component_from_text(text):
...
+def _load_component_from_yaml_or_zip_stream(stream, component_filename=None):
+    '''Loads component from a stream and creates a task factory function.
+    The stream can be YAML or a zip file with a component.yaml file inside.
+    '''
+    import zipfile
+    stream.seek(0)
+    if zipfile.is_zipfile(stream):
+        stream.seek(0)
+        with zipfile.ZipFile(stream) as zip_obj:
+            with zip_obj.open(_component_file_name) as component_streamj:
```
Comment (hongye-sun) [inline sdk/python/kfp/components/_components.py]: nit: typo? component_stream[j]

File: sdk/python/kfp/components/_components.py [hunk 2]
```diff
@@ -113,6 +113,31 @@ def load_component_from_text(text):
     return _create_task_factory_from_component_text(text, None)
 
 
+_component_file_name = 'component.yaml'
```
Comment (hongye-sun) [inline sdk/python/kfp/components/_components.py]: nit: all caps for constant name?

File: sdk/python/kfp/components/_components.py (hunk 1 above)
Comment (Ark-kun) [reply sdk/python/kfp/components/_components.py]: Thanks. Fixed.

File: sdk/python/kfp/components/_components.py (hunk 2 above)
Comment (Ark-kun) [reply sdk/python/kfp/components/_components.py]: Fixed.

**Comment by Ark-kun** [comment]
@hongye-sun , can you re-LGTM please?

**Pull request merged**